django/contrib/gis/db/models/functions.py

```diff
@@ -529,6 +529,17 @@ class Reverse(GeoFunc):
     arity = 1
 
 
+class Rotate(GeomOutputGeoFunc):
+    def __init__(self, expression, angle, x=0.0, y=0.0, **extra):
```

Comment:
Do we support all the `ST_Rotate` signatures defined [here](https://postgis.net/docs/ST_Rotate.html) or we are only implementing a couple of them for the start? It would be nice to have support for all the `ST_Rotate` functions.

Reply:
I'm struggling to implement this. I cannot pass a valid Point geometry to the Rotate function.

![image](https://github.com/user-attachments/assets/d2c87627-f9a0-4028-b307-330e9d981e54)

This line in the test function throws me a ProgrammingError "cannot adapt type 'Point' using placeholder '%t' (format: TEXT)".

I guess the problem is right here:
![image](https://github.com/user-attachments/assets/cca90d8a-17e9-4e2d-96ef-0a1b70b0945f)
 Could you help me with this?

Reply:
Sure, let me take a look and get back to you.

Reply:
@enprava I made the following changes and it worked! See below
![image](https://github.com/user-attachments/assets/22edc47b-bdb1-4279-9772-f132bc5d3e3c)
It turns out we can use the `Value` wrapper to handle non SQL-compatible fields.

Reply:
@enprava did you make any progress on this?

Reply:
@Samweli Yes. I wasn’t able to find a space of time to implement it. Thank you for your help and apologies for no replying you sooner.